sklearn_numba_dpex/common/topk.py

```diff
@@ -243,7 +289,7 @@ def _get_topk_threshold(array_in, k, group_sizes):
         work_group_size, sub_group_size = group_sizes
     else:
         work_group_size = device.max_work_group_size
-        sub_group_size = 4
+        sub_group_size = min(device.sub_group_sizes)
```

Comment:
Also I ran some profiling on this PR, and I assume that the added overhead comes mostly from `numba_dpex`, because there's one additional kernel call, + a couple more inputs to existing kernels that need unpacking. The main kernel is `create_radix_histogram_kernel` and its performance is almost unchanged.